README.md

```diff
@@ -1,15 +1,118 @@
+
+[npm-url]:https://npmjs.org/package/@moxy/next-link
+[downloads-image]:https://img.shields.io/npm/dm/@moxy/next-link.svg
+[npm-image]:https://img.shields.io/npm/v/@moxy/next-link.svg
+[build-status-url]:https://github.com/moxystudio/next-link/actions
+[build-status-image]:https://img.shields.io/github/workflow/status/moxystudio/next-link/Node%20CI/master
+[codecov-url]:https://codecov.io/gh/moxystudio/next-link
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/next-link/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/next-link
+[david-dm-image]:https://img.shields.io/david/moxystudio/next-link.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/next-link?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/next-link.svg
+
+A wrapper component that uses [next/link](https://nextjs.org/docs/api-reference/next/link) to enable client-side transitions between routes as well as external URLs.
+
+## Installation
+
+```sh
+$ npm install @moxy/next-link
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+When using Next.js, we have [next/link](https://nextjs.org/docs/api-reference/next/link) to navigate between client-side routes, but when we need to navigate to external URLs an anchor tag has to be used.
+
+This component supports both, using `NextLink` we can pass an internal or external URL, using the same component for both. With the `external` prop we tell the component which type of route is and it takes care of the rest for us, rendering the `next/link` or an `<a>` tag, maintaining the same styles between the two.
 
 ## Usage
 
-This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+```js
+import React from 'react';
+import NextLink from '@moxy/next-link';
+
+const MyPage = (props) => (
+    <div className="container">
+        <NextLink href="/about">About</NextLink>
+        <NextLink href="/contact">Contact</NextLink>
+        <NextLink external href="https://moxy.studio">MOXY</NextLink>
+    </div>
+);
+
+export default MyPage;
+```
+
+## API
+
+Besides the following supported props by the `NextLink` component, additional props will be spread to the anchor tag.
+
+#### href
+
+Type: `string` | Required: `true`
+
+The path for an internal or external URL.
+
+#### children
+
+Type: `node` | Required: `true`
+
+What to render inside the component.
+
+#### className
+
+Type: `string` | Required: `false`
+
+A className to apply to the component wrapper.
+
+#### newTab
+
+Type: `bool` | Required: `false` | Default: `false`
+
+If set to true, opens the link in a new tab.
+
+#### external
```

Comment:
can't we make this automatic by parsing `href`? or is there a situation when we need to override it?

Reply:
We probably could, but I'd argue it is best to play it safe and leave that to the responsibility of the developer. Having some kind of regex to parse that could bring us problems in the future. Do you agree?

Reply:
> We probably could, but I'd argue it is best to play it safe and leave that to the responsibility of the developer

We can still have it as an option for the user to override it.

> Having some kind of regex to parse that could bring us problems in the future. Do you agree?

What kind of problems do you mean? Edge cases we are not covering?

Reply:
I can't point one specifically, that's the thing. They may not even be edge cases but still be out of our coverage. I'd be in favour of keeping this like it is.

Are there more opinions?

Reply:
What would be our external condition? If it is a uri?

Reply:
We'll merge this as is, but [an issue](https://github.com/moxystudio/next-link/issues/2) has been opened for further discussion.

Reply:
sounds good @fsdiogo !